src/sentry/replays/consumers/recording/process_recording.py

```diff
@@ -93,6 +95,9 @@ def _process_chunked_recording(
     def submit(self, message: Message[KafkaPayload]) -> None:
         assert not self.__closed
 
+        if len(self.__futures) > self.__max_pending_futures:
+            raise MessageRejected
```

Comment:
Is it possible to add a reason to the rejection? I wonder if this will be captured upstream somehow, in which case it'll help debugging what's going on?

Reply:
The MessageRejected exception is special.  I'm just telling the producer to slow down.  The message that raised the error will be retried.  No debugging needed.